tests/unit/s2n_certificate_test.c

```diff
@@ -903,7 +903,7 @@ int main(int argc, char **argv)
                 EXPECT_SUCCESS(s2n_stuffer_copy(&input, &server->handshake.io,
                         s2n_stuffer_data_available(&input)));
 
-                EXPECT_FAILURE_WITH_ERRNO(s2n_client_cert_recv(server), S2N_ERR_CERT_INVALID);
+                EXPECT_FAILURE_WITH_ERRNO(s2n_client_cert_recv(server), S2N_ERR_DECODE_CERTIFICATE);
```

Comment:
The only behavior change here is that failing the openssl x509 parsing call returns a "decode certificate" error rather than the previous "cert invalid error". The same `d2i_X509` failure is occurring that previously occurred.
```
        server_cert = d2i_X509(NULL, &data, asn1_cert.size);
        RESULT_ENSURE(server_cert, S2N_ERR_DECODE_CERTIFICATE);
```

I'm confused by this comment on the test case
> Validate that the certificate chain we generated is parseable.

The certificate is "readable" by `s2n_x509_validator_read_asn1_cert`, but not parseable by openssl.